### ISSUE TITLE
Doorkeeper 5.1: undefined method `plaintext_token' for #<Doorkeeper::AccessGrant>

### ISSUE THREAD

**Issue by gencer** [nbulaj/doorkeeper-sequel#18]
Doorkeeper `v5.1` probably has new fields that current sequel gem hasn't yet. Below error given to me on `5.1` branch of `doorkeeper` gem.

I had to downgrade to `5.0`.

**Error:**

```
undefined method `plaintext_token' for #<Doorkeeper::AccessGrant:0x00007fe4078a4ec0>
```

What steps with sequel to follow? (Or do you plan to support this gem for 5.1?)

/cc @nbulaj

**Comment by gencer** [comment]
@nbulaj, sorry for pinging again but, doorkeeper 5.2 is on the way. Do you planning to update this gem anytime soon?

**Comment by nbulaj** [comment]
Hi @gencer . It's on the radar, but currently I don't have much time to work on it, sorry.
Maybe you wanna to produce a PR ?

**Comment by gencer** [comment]
Hi @nbulaj, the thing is, I already tried to implement this new feature. This is what i go so far:

```ruby
module DoorkeeperSequel
  module AccessGrantMixin
    extend ActiveSupport::Concern

    include SequelCompat
    include Doorkeeper::OAuth::Helpers
    include Doorkeeper::Models::Expirable
    include Doorkeeper::Models::Revocable
    include Doorkeeper::Models::Accessible
    include Doorkeeper::Models::Scopes

    included do
      plugin :validation_helpers
      plugin :timestamps

      many_to_one :application, class: 'Doorkeeper::Application'

      set_allowed_columns :resource_owner_id, :application_id,
                          :expires_in, :redirect_uri, :scopes, :code_challenge, :code_challenge_method

      def before_validation
        generate_token if new?
        super
      end
	  

      def validate
        super
        validates_presence [:resource_owner_id, :application_id,
                            :token, :expires_in, :redirect_uri]
        validates_unique [:token]
      end

      def uses_pkce?
        pkce_supported? && code_challenge.present?
      end

      def pkce_supported?
        respond_to? :code_challenge
      end	
	  
	  def plaintext_token
		if secret_strategy.allows_restoring_secrets?
		  secret_strategy.restore_secret(self, :token)
		else
		  @raw_token
		end
	  end
	  
      def secret_strategy
        ::Doorkeeper.configuration.token_secret_strategy
      end
	  
      def fallback_secret_strategy
        ::Doorkeeper.configuration.token_secret_fallback_strategy
      end
	  
    end

    module ClassMethods
      def by_token(token)
        first(token: token.to_s)
      end

      def revoke_all_for(application_id, resource_owner, clock = Time)
        where(application_id: application_id,
              resource_owner_id: resource_owner.id,
              revoked_at: nil)
          .update(revoked_at: clock.now.utc)
      end

      def generate_code_challenge(code_verifier)
        padded_result = Base64.urlsafe_encode64(Digest::SHA256.digest(code_verifier))
        padded_result.split('=')[0] # Remove any trailing '='
      end

      def pkce_supported?
        new.pkce_supported?
      end
	  
    end

    private
	
    # Generates token value with UniqueToken class.
    #
    # @return [String] token value
    #
    def generate_token
      @raw_token = UniqueToken.generate
      secret_strategy.store_secret(self, :token, @raw_token)
    end
  end
end
```

As you can see I added `plaintext_token`, `secret_strategy` and alter `generate_token`. Problem here is, it works, it fetch token successfully but authorization fails on browser and redirects to `failure`

I am trying to understand what is going wrong.

I believe there should be also `find_by_plaintext_token`


I've also tried to replicate `:token`

```
	  def plaintext_token
		self.token.to_s
	  end
```

**Comment by nbulaj** [comment]
@gencer you can compare what was changed from 5.0 to 5.1 with [this link](https://github.com/doorkeeper-gem/doorkeeper/compare/v5.0.2...v5.1.0), maybe you'll find something important that was missed

**Comment by gencer** [comment]
I've sent a PR but failed. Works here. Probably submodule needs to be updated.

**Comment by gencer** [comment]
This is fixed on PR #19.
